src/components/feedback/test/Popover-test.js

```diff
@@ -0,0 +1,316 @@
+import { mount } from 'enzyme';
+import { useRef, useState } from 'preact/hooks';
+
+import Popover, { POPOVER_VIEWPORT_HORIZONTAL_GAP } from '../Popover';
+
+function FakeComponent({ children, ...rest }) {
+  const buttonRef = useRef(null);
+  const [open, setOpen] = useState(false);
+
+  return (
+    <div className="relative">
+      <button
+        data-testid="toggle-button"
+        className="p-2"
+        ref={buttonRef}
+        onClick={() => setOpen(prev => !prev)}
+      >
+        Anchor element
+      </button>
+      <Popover open={open} anchorElementRef={buttonRef} {...rest}>
+        {open &&
+          (children ?? (
+            <>
+              Content of popover
+              <button
+                data-testid="inner-button"
+                onClick={() => setOpen(prev => !prev)}
+              >
+                Focusable element inside popover
+              </button>
+            </>
+          ))}
+      </Popover>
+    </div>
+  );
+}
+
+describe('Popover', () => {
+  let wrappers;
+  let containers;
+
+  beforeEach(() => {
+    wrappers = [];
+    containers = [];
+  });
+
+  afterEach(() => {
+    wrappers.forEach(w => w.unmount());
+    containers.forEach(c => c.remove());
+  });
+
+  function createComponent(props = {}, options = {}) {
+    const { paddingTop = 0 } = options;
+
+    const container = document.createElement('div');
+    container.style.paddingTop = `${paddingTop}px`;
+    // Add horizontal paddings to the container, so that there's room for the
+    // popover to grow if needed
```

Comment:
Yes, exactly. This padding allows to individually test cases where:

1. The popover is exactly the same size as the anchor element, because contents fit that space.
2. The popover is bigger than the anchor element, but smaller than the viewport.
3. The popover is as big as the viewport.

Without the horizontal padding, all scenarios would behave as 1 and 3 at the same time, and it would not be possible to test 2.